src/qibolab/instruments/qm.py

```diff
@@ -581,10 +581,14 @@ def fetch_results(result, ro_pulses):
                 shots = handles.get(f"{serial}_shots").fetch_all().astype(int)
             else:
                 shots = None
-            results[pulse.qubit] = results[serial] = ExecutionResults.from_components(ires, qres, shots)
+            results[pulse.qubit] = results[serial] = (
+                AveragedResults.from_components(ires, qres)
+                if average
+                else ExecutionResults.from_components(ires, qres, shots)
+            )
         return results
 
-    def play(self, qubits, sequence, nshots, relaxation_time):
+    def play(self, qubits, sequence, nshots, relaxation_time, average):
```

Comment:
Indeed this will fix the issue I pointed above, however (even though I did not explicitly test) I believe it will still fail because the `average` is not specified when the `DesignPlatform` calls `play`. The quick solution for now would be to set a default:
```suggestion
    def play(self, qubits, sequence, nshots, relaxation_time, average=True):
```
which falls back to the existing behavior.

I will try to update this later, particularly properly implement the average even when we have classified shots, which is essentially a probability, but could still be useful for some applications to perform the average (over single shots) on the device instead of getting individual shots.